leet_code_interviews/fb-2020-1-interview/938_range_sum_of_bst.js

```diff
@@ -0,0 +1,27 @@
+// 938. Range Sum of BST
+
+/**
+ * Definition for a binary tree node.
+ * function TreeNode(val) {
+ *     this.val = val;
+ *     this.left = this.right = null;
+ * }
+ */
+/**
+ * @param {TreeNode} root
+ * @param {number} L
+ * @param {number} R
+ * @return {number}
+ */
+var rangeSumBST = function(root, L, R) {
+    if (root === null) {
+        return 0;
+    } else if (root.val >= L && root.val <= R ) {
+        return root.val
+            + rangeSumBST(root.left, L, R)
+            + rangeSumBST(root.right, L, R);
+    } else {
```

Comment:
You can use some information given that the value of this root is less than the left interval or greater than the right interval to prune the left or right branch of a given BST

Reply:
bro u trolling mate

Reply:
I do not understand how you can pure here?

Reply:
@Mark5595 if you know that the value for this node is smaller than the left range, you do not have to recurse on the left child branches, because in a BST the nodes in that left branch will all be smaller than the value at the node you are at.

Reply:
Oh you right! Good catch 👍 lmk what you think now